### PR TITLE
lister admin offre

### PR DESCRIPTION
**Pull request by endricklosange**

### CODE REVIEW

**Review by maxime-guichard — DISMISSED** [review]

**Review by Erdnaxela45 — APPROVED** [review]

File: src/Entity/Company.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private string $socialReason;
+
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private int $siret;
```
Comment (LucyMerlier) [inline src/Entity/Company.php:36]: siret codes, just like postal codes or phone numbers, should be type string

File: src/Entity/Job.php [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+    /**
+     * @ORM\Column(type="string", length=200)
+     */
+    private string $post;
+
+    /**
+     * @ORM\Column(type="date")
+     */
+    private DateTimeInterface $registereAt;
```
Comment (LucyMerlier) [inline src/Entity/Job.php:30]: you missed a "d" => should be "$registeredAt"

File: src/Entity/Job.php [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+{
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     * @ORM\Column(type="integer")
+     */
+    private int $id;
+
+    /**
+     * @ORM\Column(type="string", length=200)
```
Comment (LucyMerlier) [inline src/Entity/Job.php:23]: why 200? why not 255?

File: src/Entity/Job.php [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     */
+    private string $city;
+
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private int $postalCode;
```
Comment (LucyMerlier) [inline src/Entity/Job.php:55]: postal codes, just like siret codes or phone numbers, should be type string

File: templates/admin_job/_form.html.twig [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{{ form_start(form) }}
+    {{ form_widget(form) }}
+    <button class="btn">{{ button_label|default('Save') }}</button>
```
Comment (LucyMerlier) [inline templates/admin_job/_form.html.twig:3]: so your website is gonna be in english after all?

File: templates/admin_job/_delete_form.html.twig [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<form method="post" action="{{ path('admin_job_delete', {'id': job.id}) }}" onsubmit="return confirm('Are you sure you want to delete this item?');">
+    <input type="hidden" name="_token" value="{{ csrf_token('delete' ~ job.id) }}">
+    <button class="btn">Delete</button>
+</form>
```
Comment (LucyMerlier) [inline templates/admin_job/_delete_form.html.twig:4]: so your website is gonna be in english after all?

File: templates/admin_job/edit.html.twig [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+{% block body %}
+    <h1>Edit Job</h1>
+
+    {{ include('admin_job/_form.html.twig', {'button_label': 'Update'}) }}
+
+    <a href="{{ path('admin_job_index') }}">back to list</a>
+
+    {{ include('admin_job/_delete_form.html.twig') }}
+{% endblock %}
```
Comment (LucyMerlier) [inline templates/admin_job/edit.html.twig:13]: so your website is gonna be in english after all?

File: templates/admin_job/edit.html.twig [hunk 8]
```diff
@@ -0,0 +1,13 @@
+{% extends 'base.html.twig' %}
```
Comment (LucyMerlier) [inline templates/admin_job/edit.html.twig:1]: this template extends base.html.twig directly?

File: templates/admin_job/index.html.twig [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+                </div>
+            </div>
+            <div class="row d-flex justify-content-center">
+                {% for job in jobs %}
+                    <div class="data-card text-center">
+                        <h2 class="pb-2">{{ job.Company.CompanyName|upper }}</h2>
+                        <h3 class="mt-2">{{ job.Post|upper }}</h3>
+                        <h3 class="mb-3 pb-2">{{ job.PostalCode }}
+                            {{ job.City|capitalize }}</h3>
+                        <p>{{ job.Description }}</p>
```
Comment (LucyMerlier) [inline templates/admin_job/index.html.twig:31]: should not use PascalCase here for job properties => use camelCase (ex : "job.company.companyName")

File: templates/base.html.twig [hunk 10]
```diff
@@ -9,12 +9,14 @@
...
-		{% block stylesheets %}
-			{{ encore_entry_link_tags('app') }}
-			{{ encore_entry_link_tags('navbar') }}
-		{% endblock %}
 
-		{% block javascripts %}
+        {% block stylesheets %}
+            {{ encore_entry_link_tags('app') }}
+            {{ encore_entry_link_tags('navbar') }}
+            {{ encore_entry_link_tags('adminJob') }}
```
Comment (LucyMerlier) [inline templates/base.html.twig:16]: so you will need the style for the admin job page on your whole website? this entry link should be included only on the page where you need it

File: templates/base.html.twig [hunk 11]
```diff
@@ -9,12 +9,14 @@
...
+        {% endblock %}
+		
+    {% block javascripts %}
 			{{ encore_entry_script_tags('app') }}
 		{% endblock %}
```
Comment (LucyMerlier) [inline templates/base.html.twig:21]: indentation?

File: src/Form/JobType.php [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        $builder
+            ->add('post')
+            ->add('registereAt')
+            ->add('startAt')
+            ->add('endAt')
+            ->add('hoursAWeek')
+            ->add('city')
+            ->add('postalCode')
+            ->add('description')
+            ->add('company');
```
Comment (LucyMerlier) [inline src/Form/JobType.php:23]: you should configure input types and labels here

File: src/Entity/Company.php [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+    /**
+     * @ORM\OneToMany(targetEntity=Job::class, mappedBy="company")
+     */
+    private Collection $jobs;
```
Comment (LucyMerlier) [inline src/Entity/Company.php:41]: you should do all the validations on your class properties already (using @ Assert annotations)

File: src/Entity/Job.php [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+
+    /**
+     * @ORM\Column(type="text")
+     */
+    private string $description;
+
+    /**
+     * @ORM\ManyToOne(targetEntity=Company::class, inversedBy="jobs")
+     */
+    private ?Company $company;
```
Comment (LucyMerlier) [inline src/Entity/Job.php:65]: you should do all the validations on your class properties already (using @ Assert annotations)

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

**Review by Erdnaxela45 — DISMISSED** [review]

File: migrations/Version20210610162006.php [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('CREATE TABLE company (id INT AUTO_INCREMENT NOT NULL, company_name VARCHAR(255) NOT NULL, social_reason VARCHAR(255) NOT NULL, siret INT NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('CREATE TABLE job (id INT AUTO_INCREMENT NOT NULL, company_id INT DEFAULT NULL, post VARCHAR(200) NOT NULL, registere_at DATE NOT NULL, start_at DATE NOT NULL, end_at DATE NOT NULL, hours_aweek INT NOT NULL, city VARCHAR(255) NOT NULL, postal_code INT NOT NULL, description LONGTEXT NOT NULL, INDEX IDX_FBD8E0F8979B1AD6 (company_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('ALTER TABLE job ADD CONSTRAINT FK_FBD8E0F8979B1AD6 FOREIGN KEY (company_id) REFERENCES company (id)');
```
Comment (LucyMerlier) [inline migrations/Version20210610162006.php:25]: :exclamation: you havce 2 migrations that create tables : this migration should not be here anymore :exclamation: 

=> if you wanna do modifications on your entities, either : 
 - do modifications in your entities files, and do another migration (which should then only contain "ALTER TABLE" type of requests

OR

 - do a new migration completely that will create tables correctly form the get-go (and that is what you did here), but then you HAVE TO delete the previous one (so in your case, this one)

File: src/DataFixtures/JobFixtures.php [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+            $job->setRegisteredAt(new DateTime());
+            $job->setStartAt(new DateTime());
+            $job->setEndAt(new DateTime());
+            $job->setHoursAWeek(rand(25, 35));
+            $job->setPostalCode('45');
+            $job->setCity('Ville');
+            $job->setDescription('Veniam qui aliqua deserunt do nisi consectetur pariatur consectetur tempor eiusmod.');
+            for ($c = 1; $c <= rand(1, CompanyFixtures::LOOPNUMBER); $c++) {
+                $job->setCompany($this->getReference('company_' . $c));
+            }
```
Comment (LucyMerlier) [inline src/DataFixtures/JobFixtures.php:29]: it works, but you could have done this in one line : `$job->setCompany($this->getReference('company_' . rand(1, CompanyFixtures::LOOPNUMBER)))`

File: src/Entity/Company.php [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+     * maxMessage = "Le nom de l'entreprise ne peut pas dépasser {{ limit }} caractères")
+     */
+    private string $companyName;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     * @Assert\NotBlank(message="Veuillez remplir ce champ s'il vous plait .")
+     * @Assert\Length(
+     * max=255,
+     * maxMessage = "La raison social ne peut pas dépasser {{ limit }} caractères")
```
Comment (LucyMerlier) [inline src/Entity/Company.php:38]: "raison socialE"

File: src/Entity/Company.php [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     * @ORM\Column(type="integer")
+     */
+    private int $id;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     * @Assert\NotBlank(message="Veuillez remplir ce champ s'il vous plait .")
```
Comment (LucyMerlier) [inline src/Entity/Company.php:26]: why put a period at the end of your NotBlank message when you don't end any of your other messages with a period?

File: src/Form/JobType.php [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+            ->add('city', TextType::class, [
+                'label' => 'Ville',
+                'attr' => [
+                    'placeholder' => 'Saisir une ville'
+                ]
+            ])
+            ->add('postalCode', TextType::class, [
+                'label' => 'Code postal',
+                'attr' => [
+                    'placeholder' => 'Saisir une code postal'
```
Comment (LucyMerlier) [inline src/Form/JobType.php:48]: "UN code postal"

File: templates/admin_job/_delete_form.html.twig [hunk 20]
```diff
@@ -0,0 +1,4 @@
+<form method="post" action="{{ path('admin_job_delete', {'id': job.id}) }}" onsubmit="return confirm('Are you sure you want to delete this item?');">
```
Comment (LucyMerlier) [inline templates/admin_job/_delete_form.html.twig:1]: your confirm message is still in english?

File: templates/admin_job/edit.html.twig [hunk 21]
```diff
@@ -0,0 +1,14 @@
+{% extends 'layout_admin.html.twig' %}
+
+{% block title %}Admin Modifier L'offre
+{% endblock %}
+
+{% block body %}
+	<h1>Modifier L'offre</h1>
+
+	{{ include('admin_job/_form.html.twig', {'button_label': 'Update'}) }}
```
Comment (LucyMerlier) [inline templates/admin_job/edit.html.twig:9]: you button label is in english?

File: templates/base.html.twig [hunk 22]
```diff
@@ -9,6 +9,7 @@
 			{% endblock %}
 		</title>
 
+
```
Comment (LucyMerlier) [inline templates/base.html.twig:12]: why add an empty line here?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

File: templates/admin_job/_delete_form.html.twig (hunk 20 above)
Comment (endricklosange) [reply templates/admin_job/_delete_form.html.twig:1]: c'est en francais

**Review by Erdnaxela45 — APPROVED** [review]

File: src/Form/JobType.php [hunk 23]
```diff
@@ -0,0 +1,69 @@
...
+                'label' => 'Date d\'enregistrement',
+            ])
+            ->add('startAt', DateType::class, [
+                'label' => 'Date de debut',
+            ])
+            ->add('endAt', DateType::class, [
+                'label' => 'Date de fin',
+            ])
+            ->add('hoursAWeek', NumberType::class, [
+                'label' => 'Heure par semaine',
```
Comment (LucyMerlier) [inline src/Form/JobType.php:37]: "heureS" => plural

File: src/Form/JobType.php [hunk 24]
```diff
@@ -0,0 +1,69 @@
...
+            ->add('description', TextareaType::class, [
+                'label' => 'Message',
+                'attr' => [
+                    'placeholder' => 'Votre message'
+                ]
+            ])
+            ->add('company', EntityType::class, [
+                'class' => Company::class,
+                'choice_label' => 'companyName',
+            ]);
```
Comment (LucyMerlier) [inline src/Form/JobType.php:60]: no label for this field?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

**Review by Erdnaxela45 — APPROVED** [review]

**Review by maxime-guichard — APPROVED** [review]

**Review by LucyMerlier — APPROVED** [review]

**Pull request merged**